src/buffer.c

```diff
@@ -0,0 +1,98 @@
+#include <stdlib.h>     // malloc, realloc, free
+#include <string.h>     // memset, memmove
+
+#include "buffer.h"
+
+struct zseek_buffer {
+    void *data;
+    size_t size;
+    size_t capacity;
+};
+
+zseek_buffer_t *zseek_buffer_new(size_t capacity)
+{
+    zseek_buffer_t *buffer = malloc(sizeof(*buffer));
+    if (!buffer)
+        goto fail;
+    memset(buffer, 0, sizeof(*buffer));
```

Comment:
If not absolutely necessary, better avoid memset, let the user decide.

Reply:
This does not `memset` the actual buffer, just the `struct zseek_buffer_t` handle (a few bytes, need to be initialized).

Reply:
Ah, I haven't looked close enough, just looked at the memset